hack/verify-spelling.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+export KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+# Install tools we need, but only from vendor/...
+cd ${KUBE_ROOT}
+go install ./vendor/github.com/client9/misspell/cmd/misspell
+if ! which misspell >/dev/null 2>&1; then
+    echo "Can't find misspell - is your GOPATH 'bin' in your PATH?" >&2
+    echo "  GOPATH: ${GOPATH}" >&2
+    echo "  PATH:   ${PATH}" >&2
+    exit 1
+fi
+
+# Spell checking
+# All the skipping files are defined in hack/.spelling_failures
+skipping_file="${KUBE_ROOT}/hack/.spelling_failures"
+failing_packages=$(echo `cat ${skipping_file}` | sed "s| | -e |g")
+git ls-files | grep -v -e ${failing_packages} | xargs misspell -i "Creater,creater" -error
```

Comment:
I'd prefer the use of find here, rather than ls-files piping through grep. additionally, I'd also like some logic similar to the golint script around sorting, and maintaining the known failures list.

That said, those can both be done in a follow up (to ensure that this gets merged before more spelling errors get added to the repo). This is still a worthwhile addition to the verify job. Running this locally only adds 3 seconds, but will hopefully catch a number of errors moving forward.

/lgtm

Reply:
> I'd also like some logic similar to the golint script around sorting, and maintaining the known failures list.

@cblecker We have `.spelling_failures` [file](https://github.com/kubernetes/kubernetes/pull/59463/commits/1398ac5a7d8372754ba83cc450ba06ebfd421bb2#diff-3c53f1d028902f30009db8dd9d4e02e5) here.

Reply:
Oh, can we add `-o stderr` to the flags? This will ensure that the errors get picked up in the junit report.